include/openthread/thread_ftd.h

```diff
@@ -195,6 +195,61 @@ otError otThreadSetRouterEligible(otInstance *aInstance, bool aEligible);
  */
 otError otThreadSetPreferredRouterId(otInstance *aInstance, uint8_t aRouterId);
 
+/**
+ * This enumeration represents the power supply configuration on a device.
+ *
+ * This is used as a property in `otDeviceConfig` to calculate the leader weight.
+ *
+ */
+typedef enum
+{
+    OT_POWER_SUPPLY_BATTERY           = 0, ///< Battery powered.
+    OT_POWER_SUPPLY_EXTERNAL          = 1, ///< Externally powered (mains-powered).
+    OT_POWER_SUPPLY_EXTERNAL_STABLE   = 2, ///< Stable external power with a battery backup or UPS.
+    OT_POWER_SUPPLY_EXTERNAL_UNSTABLE = 3, ///< Potentially unstable ext power (e.g. light bulb powered via a switch).
+} otPowerSupplyConfig;
+
+/**
+ * This structure represents the device configurations which are used for calculating the local leader weight on a
+ * device.
+ *
+ * The parameters are set based on device's capability, whether acting as border router, its power supply config, etc.
+ *
+ * `mIsUnstable` indicates operational stability of device and is determined via a vendor specific mechanism. It can
+ * include the following cases:
+ *  - Device internally detects that it loses external power supply more often than usual. What is usual is
+ *    determined by the vendor.
+ *  - Device internally detects that it reboots more often than usual. What is usual is determined by the vendor.
+ *
+ * When `mIsUnstable` is set to `false`, it will override `mPowerSupply` on leader weight calculations.
```

Comment:
The comment here seems not conform with the [implementation](https://github.com/openthread/openthread/pull/8670/files#diff-b5ad103de97a214f6e181a97b810e24ba67e0baeb6070ddab3701a7a2f6bfeceR118-R128) 
-  no action when `false`
-  only impact `OT_POWER_SUPPLY_EXTERNAL` and `OT_POWER_SUPPLY_EXTERNAL_STABLE` when `true`

```
   if (mIsUnstable)
    {
        switch (powerSupply)
        {
        case kPowerSupplyBattery:
        case kPowerSupplyExternalUnstable:
            break;

        default:
            weight += kIsUnstableInc;
        }
  }
```

Reply:
Thanks. I actually removed this line in new push (since we don't explain the details on weight calculation algo in the header file `thread_ftd.h` anyways).